core/trino-main/src/main/java/io/trino/execution/scheduler/ConstantPartitionMemoryEstimator.java

```diff
@@ -29,7 +30,7 @@ public MemoryRequirements getInitialMemoryRequirements(Session session, DataSize
     }
 
     @Override
-    public MemoryRequirements getNextRetryMemoryRequirements(Session session, MemoryRequirements previousMemoryRequirements, ErrorCode errorCode)
+    public MemoryRequirements getNextRetryMemoryRequirements(Session session, MemoryRequirements previousMemoryRequirements, TaskStatus failedTaskStatus, ErrorCode errorCode)
```

Comment:
Technically `TaskStatus` includes the error code, so there's some redundancy. I wonder if we really need an entire `TaskStatus`. Maybe simply pass the peak memory usage?

Reply:
Yeah - I can pass just peak memory usage as this is all we care about right now.